actions/channel_actions.ts

```diff
@@ -179,3 +192,89 @@ export function muteChannel(userId: UserProfile['id'], channelId: Channel['id'])
         mark_unread: NotificationLevels.MENTION,
     });
 }
+
+/**
+ * Fetches channels and channel members with graphql and then dispatches the result to redux store.
+ * @param teamId If team id is provided, only channels and channel members in that team will be fetched. Otherwise, all channels and all channel members will be fetched.
+ */
+export function fetchChannelsAndMembers(teamId: Team['id'] = ''): ActionFunc<{channels: ServerChannel[]; channelMembers: ChannelMembership[]}> {
+    return async (dispatch, getState) => {
+        const state = getState();
+        const currentUserId = getCurrentUserId(state);
+
+        let channelsResponse: GraphQLChannel[] = [];
+        let channelMembersResponse: GraphQLChannelMember[] = [];
+
+        try {
+            let channelsCursor = '';
+            let channelMembersCursor = '';
+            let page = 1;
+            let responsesPerPage: number;
+
+            do {
+                // eslint-disable-next-line no-await-in-loop
+                const {data, errors} = await Client4.fetchWithGraphQL<ChannelsAndChannelMembersQueryResponseType>(getChannelsAndChannelMembersQueryString(teamId, channelsCursor, channelMembersCursor));
+
+                if (errors || !data) {
+                    throw new Error(`Failed to fetch channels and channel members at page ${page}`);
+                } else if (data.channels.length === 0 || data.channelMembers.length === 0) {
```

Comment:
If we are breaking when either of them is 0, it means the length is the same, right? Then why not do the query like:
```
{
    channelMembers(userId) {
        ...
        channel {
            ...
        }
    }
}
```